src/components/SideBar/PersonalLists.vue

```diff
@@ -1,46 +1,175 @@
 <template>
   <div class="personal-lists">
-    <h1>My Lists</h1>
+    <h2>My Lists</h2>
     <div class="personal-lists-detail">
-      <div class="personal-lists-detail-left">
-        <span>icon</span>
-        <p>NameList</p>
-      </div>
-      <div class="personal-lists-detail-right">
-        <p>0</p>
-        <p>kebab_menu</p>
-      </div>
-    </div>
-    <div class="personal-lists-detail">
-      <div class="personal-lists-detail-left">
-        <span>icon</span>
-        <p>NameList2</p>
-      </div>
-      <div class="personal-lists-detail-right">
-        <p>0</p>
-        <p>kebab_menu</p>
+      <div
+        class="personal-lists-detail-element"
+        v-for="list in $lists"
+        :key="list.id"
+        :class="{
+          'list-selected': listSelected == list.id,
+        }"
+        @click="listSelected = list.id"
+      >
+        <div class="personal-lists-detail-left">
+          <span class="material-icons" id="icon">
+            list
+          </span>
+          <input
+            v-if="$edittingList === list.id"
+            :value="list.name"
+            type="text"
+            v-focus
+            @keyup.enter="editListName(list.id, $event.target.value)"
+            @blur="editListName(list.id, $event.target.value)"
+          />
+          <p class="list-name" v-else>
+            {{ list.name }}
+          </p>
+        </div>
+        <div class="personal-lists-detail-right">
+          <p>{{ countAllTasksByIdList(list.id) }}</p>
+          <span class="material-icons">
+            more_vert
+          </span>
+        </div>
       </div>
     </div>
   </div>
 </template>
 
+<script>
+import Vue from "vue";
+export default {
+  name: "PersonalLists",
+  directives: {
+    focus: {
+      inserted(el) {
+        Vue.nextTick(() => el.focus());
+      },
+    },
+  },
+  data() {
+    return {
+      listSelected: -1,
+    };
+  },
+  computed: {
+    $lists: {
+      get() {
+        return this.$store.listsData;
+      },
+    },
+    $edittingList: {
+      get() {
+        return this.$store.editting.idList;
+      },
+    },
+  },
+  methods: {
+    countAllTasksByIdList(idList) {
+      return this.$store.countAllTasksByIdList(idList);
+    },
+    editListName(idList, listName) {
+      listName = listName.toString().trim();
+      if (listName.length !== 0) {
+        const listExist = this.$store.listsData.find(
+          (list) => list.name === listName
+        );
+        if (
+          listExist === undefined ||
+          (listExist.name === listName && listExist.id === idList)
+        ) {
+          this.$store.editListName(idList, listName);
+        }
+      }
+    },
```

Comment:
- It would be better if we show warnings when the user leaves the input blank or the new list name is duplicated with the existing one.
- Don't mutate the function input.
- Irrelevant conditions should be kept separate.

E.g:

```
editListName(listId, $event) {
  const listName = $event.target.value.toString().trim();
  if (!listName) {
    return console.log("List name required!");
  }

  const duplicated = !!this.$lists.find(
    (list) => list.id !== listId && list.name === listName
  );
  if (duplicated) {
    return console.log("Duplicated name!");
  }

  this.$store.editListName(listId, listName);
}
```

Reply:
Updated!